src/Messenger/IndexationRequestHandler.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace JoliCode\Elastically\Messenger;
+
+use Elastica\Document;
+use JoliCode\Elastically\Client;
+use Psr\Log\NullLogger;
+use Symfony\Component\Messenger\Exception\UnrecoverableMessageHandlingException;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+
+abstract class IndexationRequestHandler implements MessageHandlerInterface
+{
+    const OP_INDEX = 'index';
+    const OP_DELETE = 'delete';
+    const OP_UPDATE = 'update';
+    const OP_CREATE = 'create';
+
+    const OPS = [
+        self::OP_INDEX,
+        self::OP_DELETE,
+        self::OP_UPDATE,
+        self::OP_CREATE,
+    ];
+
+    private $client;
+
+    public function __construct(Client $client)
+    {
+        $this->client = $client;
+
+        // Disable the logs for memory concerns
+        $this->client->setLogger(new NullLogger());
+    }
+
+    public function __invoke(IndexationRequest $message)
+    {
+        $indexer = $this->client->getIndexer();
+        $indexToClass = $this->client->getConfig(Client::CONFIG_INDEX_CLASS_MAPPING);
+        $indexName = array_search($message->getType(), $indexToClass, true);
+
+        if (!$indexName) {
+            throw new UnrecoverableMessageHandlingException(sprintf('The given type (%s) does not exist!', $message->getType()));
+        }
+
+        if (self::OP_DELETE === $message->getOp()) {
+            $indexer->scheduleDelete($indexName, $message->getId());
+            $indexer->flush();
+
+            return;
+        }
+
+        $model = $this->fetchModel($message->getType(), $message->getId());
+
+        if (!$model) {
+            // ID does not exists, delete
```

Comment:
Why?

Reply:
To me, if `fetchModel` fail to found the entity/the data, that means it does not exists. If it did exists before, triggering a delete is a safe way to sync the `fetchModel` behavior with what we have in Elasticsearch.

Reply:
But if the does not exist, if you delete it you basically delete nothing.